src/Symfony/Component/Scheduler/Trigger/CallbackMessageProvider.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Scheduler\Trigger;
+
+use Symfony\Component\Scheduler\Generator\MessageContext;
+
+final class CallbackMessageProvider implements MessageProviderInterface, \Stringable
+{
+    private \Closure $callback;
+
+    public function __construct(callable $callback, private string $description = '')
+    {
+        $this->callback = $callback(...);
+    }
+
+    public function getMessages(MessageContext $context): iterable
+    {
+        return ($this->callback)($context);
+    }
+
+    public function __toString(): string
```

Comment:
Why do we need the message provider to be stringable ? And if we actually need it, why isn't it enforced in the interface ?

Reply:
Also, if the goal is to provide a unique id, I would rather not call that `description` (and I would probably rather use a `getId` method)

Reply:
It is because of https://github.com/symfony/symfony/blob/efd41f476df0e9ec73804c8b482046cac8021f6f/src/Symfony/Component/Scheduler/RecurringMessage.php#L95

In which I just tried to keep the existing behaviour where possible, even though I don't fully understand the need for a duplication-check like this.

The `getId` method is fine by me.